### PR TITLE
Added Comprehensive Examples for set Function in Documentation

### LINKED ISSUE #5031: add examples of `set` function in documentation

### PR DESCRIPTION
**Pull request by Nj221102**
## Description:
This PR enhances the documentation for the `set` function in the `data.table` package by adding detailed examples. These examples demonstrate the usage of `set` for assigning values to multiple columns and handling multiple values efficiently. The aim is to provide clear and comprehensive usage scenarios, thereby improving user understanding and ensuring best practices.

## Changes Made:
1. Added detailed examples to the `set` function documentation.
2. Demonstrated assignment to multiple columns of multiple values.

closes #5031

### CODE REVIEW

**Comment by tdhock** [comment]
Would be good to add an example using `set(DT, j=some_name, value=some_value)` which is how I typically use set (without i argument)

**Comment by Nj221102** [comment]
> Would be good to add an example using `set(DT, j=some_name, value=some_value)` which is how I typically use set (without i argument)
Sure :)

File: man/assign.Rd [hunk 1]
```diff
@@ -140,6 +140,18 @@ sq_col_idx = grep('d$', names(DT))
...
 
+# Examples using `set` function
+## Set value for single cell
+set(DT, 1L, "b", 10L)
+## Set values for multiple columns in a specific row
+set(DT, 2L, c("b", "d"), list(20L, 30L))
+## Set values by column indices
+set(DT, 3L, c(2L, 4L), list(40L, 50L))
+## Set value for an entire column without specifying rows
+set(DT, j = "b", value = 100L)
```
Comment (MichaelChirico) [inline man/assign.Rd:151]:
```suggestion
set(DT, j = "b", value = 100L)
set(DT, NULL, "b", 100L) # equivalent
```

**Comment by MichaelChirico** [comment]
LGTM. Should there be an example with >1 (but less than `nrow`) rows being assigned? Or is that obvious enough from the existing example?

**Comment by Nj221102** [comment]
> LGTM. Should there be an example with >1 (but less than `nrow`) rows being assigned? Or is that obvious enough from the existing example?

Let me just add that as well, more examples will help people understand better, even though its pretty obvious but there is no harm in one  more example i think.

**Comment by Nj221102** [comment]
> LGTM. Should there be an example with >1 (but less than `nrow`) rows being assigned? Or is that obvious enough from the existing example?

done :)

**Review by MichaelChirico — APPROVED** [review]
Great, thanks!

**Pull request merged**